beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -709,22 +730,39 @@ impl<E: EthSpec> PeerManager<E> {
                     "peer_id" => %peer_id, "new_seq_no" => meta_data.seq_number());
             }
 
-            let custody_subnet_count_opt = meta_data.custody_subnet_count().copied().ok();
+            let custody_group_count_opt = meta_data.custody_group_count().copied().ok();
             peer_info.set_meta_data(meta_data);
 
             if self.network_globals.spec.is_peer_das_scheduled() {
                 // Gracefully ignore metadata/v2 peers. Potentially downscore after PeerDAS to
                 // prioritize PeerDAS peers.
-                if let Some(custody_subnet_count) = custody_subnet_count_opt {
-                    match self.compute_peer_custody_subnets(peer_id, custody_subnet_count) {
-                        Ok(custody_subnets) => {
+                if let Some(custody_group_count) = custody_group_count_opt {
+                    match self.compute_peer_custody_groups(peer_id, custody_group_count) {
+                        Ok(custody_groups) => {
+                            let custody_subnets = custody_groups
+                                .into_iter()
+                                .flat_map(|custody_index| {
+                                    self.subnets_by_custody_group
+                                        .get(&custody_index)
+                                        .cloned()
+                                        .unwrap_or_else(|| {
+                                            warn!(
+                                                self.log,
+                                                "Custody group not found in subnet mapping";
+                                                "custody_index" => custody_index,
+                                                "peer_id" => %peer_id
+                                            );
```

Comment:
This should never happen as we compute `subnets_by_custody_group` ourselves. Is it necessary to have a warn here?

Reply:
Yes agree that this _should_ not happen - we also validate peer's `custody_group_count` when computing custody groups.

What error handling are you suggesting? I don't think we should panic, or just swallow the error _if_ it happens in case of a bug or invalid input that somehow slip through our validation.